### ISSUE TITLE
how to prepare text for single label text classification?

### ISSUE THREAD

**Issue by Gautam18k12** [zalandoresearch/flair#571]
For instance, there is provision to prepare corpus for multi label classification like __Label__label1 __Label__label2 "Text" . But if we have only one label either label=1 or label= 0 . in this case how can i prepare text for corpus.  

Let say i have a label called "XYZ".. i have prepared text like __Label__XYZ "Text"(where label=1)  __Label__non_XYZ(where label=0). Is this right approach to prepare text?

**Comment by alanakbik** [comment]
Hi @Gautam18k12 in this case you only put one label per line, i.e. always the label the line should be classified with. Also make sure you set multi_label=False during training!

**Comment by alanakbik** [comment]
Hope this answered the question! Closing the issue, but feel free to reopen if you have further questions.

**Comment by Gautam18k12** [comment]
> Hi @Gautam18k12 in this case you only put one label per line, i.e. always the label the line should be classified with. Also make sure you set multi_label=False during training!

Hi , As said i have done same thing .. but when i doing prediction , Sentences getting tagged with one of the label i.e. "xyz" or "non_xyz" with some probablities. Here are the prediction for some sentences.

[non_xyz (1.0)]
[non_xyz (0.005929766222834587)]
[non_xyz (0.9034425616264343)]
[xyz (0.6940355896949768)]
[non_xyz (0.20091956853866577)]
[non_xyz (0.6414812803268433)]
[xyz (0.8133372664451599)]
[non_xyz (0.4464661180973053)]
[xyz (1.0)]

Now, the issue is that prediction probabilities always should be >=0.5 for a label to tagged with particular sentences. But you can notice that probabilities are very smaller than 0.5 for a label which flair does prediction for example  **[non_xyz (0.005929766222834587)]** or **[non_xyz (0.005929766222834587)]**.

So my question is how does it decide about probability and prediction ?
and how the prediction work. Even i peeped into sources code function **trainer.model.forward( )**. here what i understand it that a label will be tagged only if confidence>=0.5.

**Comment by alanakbik** [comment]
Hi @Gautam18k12 that is strange - I don't think it should tag anything with confidence lower than 0.5. This could be an error in the single-label classification method. We need to take a closer look at this. Thanks for reporting this!

**Comment by ebbotai** [comment]
@alanakbik - Do you have an update on this matter? 
I have train a model with single label classification. The training file contains 500 rows, 100 in test and 100 in dev. 

The model scores 1.0
`payment    tp: 64 - fp: 0 - fn: 0 - tn: 0 - precision: 1.0000 - recall: 1.0000 - accuracy: 1.0000 - f1-score: 1.0000
`

The problem is that whatever I type when predict it match it against the payment label with the score 1.0.

In the training procedure I have tried both DocumentPoolEmbeddings and DocumentRNNEmbeddings with the following parameters
`max_epochs=150, learning_rate=0.05, mini_batch_size = 8`
